R/snapshot.R

```diff
@@ -885,40 +885,40 @@ renv_snapshot_filter <- function(project, records, type, packages, exclude) {
   type <- aliases[[type]] %||% type
 
   result <- switch(type,
-    all      = renv_snapshot_filter_all(project, records),
-    custom   = renv_snapshot_filter_custom(project, records),
-    explicit = renv_snapshot_filter_explicit(project, records),
-    implicit = renv_snapshot_filter_implicit(project, records),
-    packages = renv_snapshot_filter_packages(project, records, packages),
+    all      = renv_snapshot_filter_all(project, records, exclude),
+    custom   = renv_snapshot_filter_custom(project, records, exclude),
+    explicit = renv_snapshot_filter_explicit(project, records, exclude),
+    implicit = renv_snapshot_filter_implicit(project, records, exclude),
+    packages = renv_snapshot_filter_packages(project, records, packages, exclude),
     stopf("unknown snapshot type '%s'", type)
   )
 
-  if (length(exclude))
-    result <- exclude(result, exclude)
-
   result
 
 }
 
-renv_snapshot_filter_all <- function(project, records) {
-  renv_snapshot_filter_impl(project, records, names(records), "all")
+renv_snapshot_filter_all <- function(project, records, exclude) {
+  renv_snapshot_filter_impl(project, records, names(records), "all", exclude)
 }
 
-renv_snapshot_filter_impl <- function(project, records, packages, type) {
+renv_snapshot_filter_impl <- function(project, records, packages, type, exclude) {
 
   # make sure we include renv
   packages <- unique(c(packages, "renv"))
 
   # warn if some required packages are missing
   ignored <- c(renv_project_ignored_packages(project), renv_packages_base())
-  missing <- setdiff(packages, c(names(records), ignored))
+  missing <- setdiff(packages, c(names(records), ignored, exclude))
   if (!`_renv_status_running`)
     renv_snapshot_filter_report_missing(missing, type)
 
   # ignore packages as defined by project
   used <- setdiff(packages, ignored)
 
   # include transitive dependencies
+  #
+  # TODO: what if a package that a user requested be ignored pops up here; should
```

Comment:
IMO, I think it'd be sufficient to document that `exclude` won't exclude transitive dependencies.